src/target/arc_ocd.c

```diff
@@ -141,6 +141,13 @@ int arc_ocd_deassert_reset(struct target *target)
 	/* deassert reset lines */
 	jtag_add_reset(0, 0);
 
+	if (!target->reset_halt && target->state == TARGET_RESET)
```

Comment:
You need to add a comment why this is needed: that ARC target after reset might be in the **halt** state, so if script requested processor to resume, then it must be manually started to ensure that this request is satisfied,